apps/src/javalab/JavalabEditor.jsx

```diff
@@ -53,6 +75,41 @@ class JavalabEditor extends React.Component {
         <PaneHeader hasFocus={true}>
           <PaneSection>Editor</PaneSection>
         </PaneHeader>
+        <div>
+          {this.state.renameFileActive ? (
+            <div style={style.tabs}>
+              <div style={style.tab}>
+                <input
+                  type="text"
+                  ariaLabel="file name"
+                  value={this.props.fileName}
+                  onChange={e => this.props.setFileName(e.target.value)}
+                  onBlur={() => this.setState({renameFileActive: false})}
+                />
+              </div>
+              <button
+                type="button"
+                onClick={() => this.setState({renameFileActive: false})}
+                className="btn btn-default btn-sm"
+                style={style.button}
+              >
+                Save
+              </button>
+            </div>
+          ) : (
```

Comment:
It seems like this might be a good spot to start adding testing. i.e. do you expect that with the final product we'd have either a button or an input and a button, dependent on the state? Similarly, are some of the redux functionalities at the point where testing makes sense?

Reply:
the final design is going to have a dropdown menu with options `rename` and `delete`, so I don't know if it's worth testing at this point. Redux might be worth adding testing, although I'm not sure how tied to the UI functionality the redux testing would be.

Reply:
meta thought: the redux functions are quite simple, I'm not sure unit testing them independently would give us all that much helpful information beyond they haven't been changed.